clash-protocols-base/src/Protocols/Internal/Classes.hs

```diff
@@ -134,6 +138,170 @@ types:
 newtype Circuit a b
   = Circuit ((Fwd a, Bwd b) -> (Bwd a, Fwd b))
 
+-- | Conversion from booleans to protocol specific acknowledgement values.
+class (Protocol a) => Backpressure a where
+  -- | Interpret list of booleans as a list of acknowledgements at every cycle.
+  -- Implementations don't have to account for finite lists.
+  boolsToBwd :: Proxy a -> [Bool] -> Bwd a
+
+--------------------------------- SIMULATION -----------------------------------
+
+{- | Specifies option for simulation functions. Don't use this constructor
+directly, as it may be extend with other options in the future. Use 'def'
+instead.
+-}
+data SimulationConfig = SimulationConfig
```

Comment:
Hmmm types are not classes either; I see I made the same transgression with `Circuit`. I suggest we move `Circuit` back to `Protocols.Internal`. It was in all likelihood a silly mistake I made.